### PR TITLE
[OFFAPPS-1164] If setting has any value, use it; show error

### PR DESCRIPTION
**Pull request by ocke**
## Description
I found that when the setting is incorrect (i.e. a string with letters, 'abc') the app would continue to try to use the requirement field.

This should not be the case. If there is a setting, we should *always* use that setting. If the setting is wrong, let the user know.

When there is an error, display the error

**Go by commit for easy reviewing**

## References
[JIRA ticket](https://zendesk.atlassian.net/browse/OFFAPPS-1164)

## CCs
@zendesk/apps-migration
@zendesk/localization 

## Risks
* [low] Field not working for linked ticket

### CODE REVIEW

File: spec/src/app_spec.js [hunk 1]
```diff
@@ -950,3 +951,97 @@ describe('App', () => {
...
+    describe('when setting is undefined', () => {
+      beforeEach(() => {
+        appInstance._metadata.settings.ancestry_field = undefined
+      })
+
+      it('continue with requirement field', (done) => {
+        appInstance.getAncestryFieldId().then((ancestryFieldId) => {
+          expect(ancestryFieldId).toEqual(456)
+          done()
+        }).catch(console.error)
```
Comment (keitheous) [inline spec/src/app_spec.js]: 🤔 what does this do?

Do we expect to see error being console while running `npm run test`

File: spec/src/app_spec.js [hunk 2]
```diff
@@ -193,7 +188,7 @@ describe('App', () => {
     describe('when ticket is not closed', () => {
       it('calls "#loadIfDataReady"', (done) => {
         spyOn(client, 'get').and.returnValue(Promise.resolve({'ticket': {'status': 'open'}}))
-        spyOn(appInstance, 'checkForms').and.returnValue(Promise.resolve())
+        spyOn(appInstance, 'checkForm').and.returnValue(Promise.resolve())
```
Comment (keitheous) [inline spec/src/app_spec.js]: In this PR there is a number of `returnValue(Promise.resolve())`, curious what you think about switching this to 
`jest.fn().mockResolvedValue()` instead. I am certainly considering this for redaction app.

[Documentation here](https://jestjs.io/docs/en/mock-function-api.html#mockfnmockresolvedvaluevalue)

File: spec/src/app_spec.js [hunk 3]
```diff
@@ -58,13 +58,13 @@ describe('App', () => {
 
     describe('when forms are invalid', () => {
       it('throws an error', () => {
-        appInstance.checkForms.and.returnValue(rejectedPromise)
+        appInstance.checkForm.and.returnValue(rejectedPromise)
```
Comment (keitheous) [inline spec/src/app_spec.js]: similar to the comment below 👇 

[jest.fn().mockResolvedValue()](https://jestjs.io/docs/en/mock-function-api.html#mockfnmockrejectedvaluevalue)

File: spec/src/app_spec.js (hunk 1 above)
Comment (ocke) [reply spec/src/app_spec.js]: Oops, I will delete this

File: spec/src/app_spec.js (hunk 2 above)
Comment (ocke) [reply spec/src/app_spec.js]: This isn't jest 😆

Comment (keitheous) [reply spec/src/app_spec.js]: Sorry this is not Jest

File: spec/src/app_spec.js (hunk 3 above)
Comment (keitheous) [reply spec/src/app_spec.js]: Sorry this is not Jest

File: src/javascript/app.js [hunk 4]
```diff
@@ -134,33 +136,25 @@ const App = {
...
-      })
-
-      if (formErrors.length) {
-        const error = new Error(this.I18n.t('linked_data_field_missing'))
-        error.subErrors = formErrors
-        throw error
+  checkForm: function () {
+    return this.getAncestryFieldId().then((ancestryFieldId) => {
+      return Promise.all([
+        this.getAncestryFieldId(),
```
Comment (hoangtrvu) [inline src/javascript/app.js]: Why do we get this twice (L146 and here)?

File: src/javascript/app.js [hunk 5]
```diff
@@ -17,6 +17,8 @@ const App = {
     // APP EVENTS
     'app.created': 'onCreated',
     'ticket.status.changed': 'onTicketStatusChanged',
+    'ticket.form.id.changed': 'onTicketFormChanged',
```
Comment (hoangtrvu) [inline src/javascript/app.js]: `onTicketFormIDChanged` instead?

**Pull request merged**